apps/snitch_core/test/tools/helpers/taxonomy_test.exs

```diff
@@ -0,0 +1,50 @@
+defmodule Snitch.Tools.Helpers.Taxonomy do
+  use ExUnit.Case, async: false
+  use Snitch.DataCase
+
+  alias Snitch.Tools.Helper.Taxonomy, as: TaxonomyHelper
+  alias Snitch.Domain.Taxonomy, as: TaxonomyDomain
+  alias Snitch.Repo
+
+  @brands_taxonomy {"Brands",
+                    [
+                      {"Bags", []},
+                      {"Mugs", []},
+                      {"Clothing",
+                       [
+                         {"Shirts", []},
+                         {"T-Shirts", []}
+                       ]}
+                    ]}
+
+  describe "create_taxonomy/1" do
+    test "create valid taxonomy" do
+      taxonomy =
+        TaxonomyHelper.create_taxonomy(@brands_taxonomy)
+        |> Repo.preload(:root)
+
+      taxons = get_taxons_name_list(taxonomy.root)
+      assert taxons == ["Brands", "Bags", "Mugs", "Clothing", "Shirts", "T-Shirts"]
+    end
+
+    test "create single node taxonomy" do
+      taxonomy =
+        TaxonomyHelper.create_taxonomy({"Root", []})
+        |> Repo.preload(:root)
+
+      taxons = get_taxons_name_list(taxonomy.root)
+      assert taxons == ["Root"]
+    end
+  end
+
+  defp get_taxons_name_list(taxon) do
```

Comment:
This function can move to the helper module as well. Though the utility of the flat list of names is rather limited.